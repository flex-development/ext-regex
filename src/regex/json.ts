/**
 * @file Regular EXPRESSIONS - EXT_JSON_REGEX
 * @module ext-regex/regex/json
 */

/**
 * JSON file extension regex.
 *
 * Supported extensions:
 *
 * - `.json`
 * - `.json5`
 * - `.jsonc`
 *
 * Captured groups:
 *
 * - `type`: Character after `'json'` in file extension
 *
 * @see https://regex101.com/r/YVxapd
 *
 * @const {RegExp} EXT_JSON_REGEX
 */
const EXT_JSON_REGEX: RegExp = /\.json(?<type>[5c])?(?=\s*$)/

export default EXT_JSON_REGEX
