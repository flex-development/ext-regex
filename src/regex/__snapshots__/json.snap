// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:EXT_JSON_REGEX > .json > should match extension ".json" 1`] = `
{
  "0": ".json",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .json > should match module id given last extension is ".json" 1`] = `
{
  "0": ".json",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .json > should match module id given only extension is ".json" 1`] = `
{
  "0": ".json",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .json5 > should match extension ".json5" 1`] = `
{
  "0": ".json5",
  "1": "5",
  "groups": {
    "type": "5",
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .json5 > should match module id given last extension is ".json5" 1`] = `
{
  "0": ".json5",
  "1": "5",
  "groups": {
    "type": "5",
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .json5 > should match module id given only extension is ".json5" 1`] = `
{
  "0": ".json5",
  "1": "5",
  "groups": {
    "type": "5",
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .jsonc > should match extension ".jsonc" 1`] = `
{
  "0": ".jsonc",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .jsonc > should match module id given last extension is ".jsonc" 1`] = `
{
  "0": ".jsonc",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_JSON_REGEX > .jsonc > should match module id given only extension is ".jsonc" 1`] = `
{
  "0": ".jsonc",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;
